### PR TITLE
Make ServerHandshake::ActionMoveVisitor an implementation detail

### PR DESCRIPTION
**Pull request by deadalnix**
ActionMoveVisitor is fizz specific, so we want to remote it from the public API.

### CODE REVIEW

File: quic/server/handshake/ServerHandshake.cpp [hunk 1]
```diff
@@ -350,132 +298,183 @@ void ServerHandshake::processPendingEvents() {
...
-}
+ private:
+  ServerHandshake& server_;
+};
 
-void ServerHandshake::ActionMoveVisitor::operator()(
-    fizz::server::AttemptVersionFallback&) {
-  CHECK(false) << "Fallback Unexpected";
-}
+void ServerHandshake::processActions(
```
Comment (yangchi) [inline quic/server/handshake/ServerHandshake.cpp:430]: was this changed at all or was it simpled moved to this location?

Comment (deadalnix) [reply quic/server/handshake/ServerHandshake.cpp:430]: It was simply moved. It is using ActionMoveVisitor and therefore must be defined after. The code is identical at on small exception.

File: quic/server/handshake/ServerHandshake.cpp [hunk 2]
```diff
@@ -350,132 +298,183 @@ void ServerHandshake::processPendingEvents() {
...
+    fizz::server::ServerStateMachine::CompletedActions actions) {
+  // This extra DestructorGuard is needed due to the gap between clearing
+  // actionGuard_ and potentially processing another action.
+  folly::DelayedDestruction::DestructorGuard dg(conn_);
 
-void ServerHandshake::ActionMoveVisitor::operator()(fizz::EndOfData&) {
-  server_.onError(std::make_pair(
-      "Unexpected close notify received", TransportErrorCode::INTERNAL_ERROR));
-}
+  ActionMoveVisitor visitor(*this);
```
Comment (deadalnix) [inline quic/server/handshake/ServerHandshake.cpp:436]: This is what was changed. The visitor used to be a field of the handshake and now it is a local variable. Everything else is the same.

**Comment by xttjsn** [comment]
lgtm

**Pull request closed without merging**